Lib/fractions.py

```diff
@@ -414,27 +434,39 @@ def __str__(self):
         else:
             return '%s/%s' % (self._numerator, self._denominator)
 
-    def __format__(self, format_spec, /):
-        """Format this fraction according to the given format specification."""
-
-        # Backwards compatiblility with existing formatting.
-        if not format_spec:
-            return str(self)
+    def _format_int_style(self, match):
+        """Helper method for __format__; handles 'd' presentation type."""
 
         # Validate and parse the format specifier.
-        match = _FLOAT_FORMAT_SPECIFICATION_MATCHER(format_spec)
-        if match is None:
-            raise ValueError(
-                f"Invalid format specifier {format_spec!r} "
-                f"for object of type {type(self).__name__!r}"
-            )
-        elif match["align"] is not None and match["zeropad"] is not None:
-            # Avoid the temptation to guess.
-            raise ValueError(
-                f"Invalid format specifier {format_spec!r} "
-                f"for object of type {type(self).__name__!r}; "
-                "can't use explicit alignment when zero-padding"
-            )
+        fill = match["fill"] or " "
+        align = match["align"] or ">"
+        pos_sign = "" if match["sign"] == "-" else match["sign"]
+        alternate_form = bool(match["alt"])
+        minimumwidth = int(match["minimumwidth"] or "0")
+        thousands_sep = match["thousands_sep"] or ''
+
+        # Determine the body and sign representation.
+        n, d = self._numerator, self._denominator
+        if d > 1 or alternate_form:
+            body = f"{abs(n):{thousands_sep}}/{d:{thousands_sep}}"
```

Comment:
I think that it may be optimized, taking into account that almost in all cases `sign` is followed by `body`, and when the padding is inserted between them, you can write `body[0] + padding + body[1:]`, but let left it to the following PR, since there is also a couple of special cases depending on `match["sign"]`.

Since in all cases, except `align == "="`, `sign` is followed by `body`, it may be

Reply:
Yes, I didn't worry too much about performance here. I'm happy to make improvements in a followup PR, so long as we get the semantics right for this one.